### PR TITLE
Remove `@EnableDiscoveryClient` as it is optional

### LINKED ISSUE #146: Remove @EnableDiscoveryClient
`@EnableDiscoveryClient` is now optional.
See https://github.com/spring-cloud/spring-cloud-commons/issues/218

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Fixes #146.

### CODE REVIEW

File: spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java [hunk 1]
```diff
@@ -42,8 +42,8 @@
 @ConditionalOnMissingBean(type = "org.springframework.cloud.zookeeper.discovery.ZookeeperLifecycle")
 @ConditionalOnZookeeperDiscoveryEnabled
 @ConditionalOnProperty(value = "spring.cloud.service-registry.auto-registration.enabled", matchIfMissing = true)
-@AutoConfigureAfter(ZookeeperServiceRegistryAutoConfiguration.class)
-@AutoConfigureBefore( {AutoServiceRegistrationAutoConfiguration.class, ZookeeperDiscoveryAutoConfiguration.class} )
+@AutoConfigureAfter( {AutoServiceRegistrationAutoConfiguration.class,ZookeeperServiceRegistryAutoConfiguration.class} )
```
Comment (spencergibb) [inline spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: Why this change?

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: Because now the `AutoServiceRegistrationProperties` bean will only be created once `AutoServiceRegistrationAutoConfiguration` is run.  See https://github.com/spring-cloud/spring-cloud-commons/pull/245/files#diff-ead0de8e4600f43bcaadeee88d39b828R16

Comment (spencergibb) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: I don't see how that justifies mocinv from Before to After. `AutoServiceRegistrationAutoConfiguration` will fail with an exception because of this change.

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: auto configuration boggles my mind.  The exception will only be thrown is `failfast` is set to true, which may explain why I dont see any exceptions being thrown.  However even with telling it to auto configure after `AutoServiceRegistrationAutoConfiguration`, it is still running before!  What?!

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: @spencergibb do u know why `@ConditionalOnBean(AutoServiceRegistrationProperties.class)` is even on this class?  `AutoServiceRegistrationProperties` isn't used anywhere

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: @spencergibb ping ^^^

Comment (spencergibb) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: It's the marker bean that auto-registration was enabled

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: That seems a little redundant since there is already a `@ConditionalOnProperty` for `spring.cloud.service-registry.auto-registration.enabled`.

Comment (ryanjbaxter) [reply spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/serviceregistry/ZookeeperAutoServiceRegistrationAutoConfiguration.java]: @spencergibb do you think we can remove the `@ConditionalOnBean(AutoServiceRegistrationProperties.class)` and keep the `@ConditionalOnProperty(spring.cloud.service-registry.auto-registration.enabled)`?
